### PR TITLE
refactor: refact the setup spec mount

### PR DESCRIPTION
**Pull request by rudyfly**
### Ⅰ. Describe what this PR did
refect the setup spec mount, make it a logical structure as below:
1. override the default spec mounts by container's config;
2. setup container's mounts into spec mounts;
3. setup or modify extra configs into spec mounts;
4. sort all spec mounts.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
NO

### Ⅳ. Describe how to verify it
NA

### Ⅴ. Special notes for reviews

Signed-off-by: Rudy Zhang <rudyflyzhang@gmail.com>

### CODE REVIEW

**Comment by allencloud** [comment]
Could we separate the function `setupMounts` with more readable functions? 
Every function contains the detailed implementation of the code block.

I think it is too long and a little bit difficult to understand. @rudyfly @Ace-Tang

File: daemon/mgr/spec_mount.go [hunk 1]
```diff
@@ -42,7 +42,12 @@ func clearReadonly(m *specs.Mount) {
 
 // setupMounts create mount spec.
 func setupMounts(ctx context.Context, c *Container, s *specs.Spec) error {
+	if c.HostConfig == nil {
+		return nil
```
[the remark this answers is not in the document]
Comment (HusterWan) [reply daemon/mgr/spec_mount.go]: delete this line

**Comment by rudyfly** [comment]
PTAL @Ace-Tang @HusterWan @allencloud

File: daemon/mgr/spec_mount.go [hunk 2]
```diff
@@ -121,21 +112,54 @@ func setupMounts(ctx context.Context, c *Container, s *specs.Spec) error {
...
 	}
 
-	s.Mounts = sortMounts(mounts)
+	return mounts, nil
+}
+
+// setupMounts create mount spec.
+func setupMounts(ctx context.Context, c *Container, s *specs.Spec) error {
+	var (
+		mounts []specs.Mount
```
Comment (HusterWan) [inline daemon/mgr/spec_mount.go]: no need to define this parameter, WDYT?

File: daemon/mgr/spec_mount.go [hunk 3]
```diff
@@ -40,10 +41,7 @@ func clearReadonly(m *specs.Mount) {
 	m.Options = opts
 }
 
-// setupMounts create mount spec.
-func setupMounts(ctx context.Context, c *Container, s *specs.Spec) error {
-	var mounts []specs.Mount
-	// Override the default mounts which are duplicate with user defined ones.
+func overrideDefaultMount(mounts []specs.Mount, c *Container, s *specs.Spec) ([]specs.Mount, error) {
```
Comment (HusterWan) [inline daemon/mgr/spec_mount.go]: Maybe we can add some test cases for the new function.

File: daemon/mgr/spec_mount.go (hunk 2 above)
Comment (rudyfly) [reply daemon/mgr/spec_mount.go]: I think it need.

File: daemon/mgr/spec_mount.go (hunk 3 above)
Comment (rudyfly) [reply daemon/mgr/spec_mount.go]: It isn't complicated logic, I think it no need to add unit test.

Comment (Ace-Tang) [reply daemon/mgr/spec_mount.go]: Should add, but can be in another pr

**Comment by Ace-Tang** [comment]
LGTM

**Pull request merged**